src/libcore/option.rs

```diff
@@ -981,6 +981,107 @@ impl<T> From<T> for Option<T> {
     }
 }
 
+// The Option<T> type used to be defined as { None, Some(T) }, but for codegen
+// reasons we reversed it in #49499 to reduce the amount of work that needs
+// to be done for Result<T, ()> <-> Option<T> conversions. Keeping the derived
+// Ord and PartialOrd implementations would make that swap a breaking change.
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<T> Ord for Option<T>
+where
+    T: Ord,
+{
+    #[inline]
+    fn cmp(&self, other: &Self) -> cmp::Ordering {
+        let self_discr = unsafe { intrinsics::discriminant_value(self) } as isize;
```

Comment:
Couldn't this use `is_none()` or `is_some()`?

Reply:
Yeah I guess I could use `is_none`, I'll check that it still produces the same code as for 
`Result<T, ()>`.